livekit-rtc/livekit/rtc/video_stream.py

```diff
@@ -13,14 +13,22 @@
 # limitations under the License.
 
 import asyncio
+from dataclasses import dataclass
 from typing import Optional
 
 from ._ffi_client import FfiHandle, FfiClient
 from ._proto import ffi_pb2 as proto_ffi
 from ._proto import video_frame_pb2 as proto_video_frame
 from ._utils import RingQueue, task_done_logger
 from .track import Track
-from .video_frame import VideoFrame, VideoFrameBuffer
+from .video_frame import VideoFrame
+
+
+@dataclass
+class VideoFrameReceived:
```

Comment:
Makes sense. I'm wondering if maybe VideoFrame is really "VideoBuffer" and "VideoFrameReceived" can be "VideoFrame"?

Reply:
I did that because AudioFrame is not AudioBuffer.
So I wanted to keep consistency
Otherwise we will have to do some breaking changes

Reply:
so conceptually, frame is the container for video data, and there's another construct needed to contain that element plus all of the other data.. we would have the same thing for audio too right? in terms of passing along the timestamps?

how about calling it `VideoSample` and `AudioSample`.. A sample contains a Frame plus other information about how that frame should be rendered.. when publishing you capture frames and pass along those attributes.

maybe it should be called `capture` or `push` instead of `capture_frame`